Sources/Tagged/Tagged.swift

```diff
@@ -35,7 +35,12 @@ extension Tagged: Comparable where RawValue: Comparable {
 
 extension Tagged: Decodable where RawValue: Decodable {
   public init(from decoder: Decoder) throws {
-    self.init(rawValue: try .init(from: decoder))
+    do {
+      self.init(rawValue: try decoder.singleValueContainer().decode(RawValue.self))
```

Comment:
@buscarini Can you explain the difference between going through the container vs straight from decoder?

Im working on a related problem: with Pointfreeco's ArbitraryDecoder, if I go through the container I can customize whats being decoded (which I need e.g. for String based enums, which almost certainly fail from an arbitrary string).

But it would help if I understood what the difference between the two is for your usecase, so that I dont introduce unexpected bugs.

Thanks

Reply:
Specifically, I consider going through the container to be the correct way (it gives the decoder a chance to do whatever it needs to do). So I wonder which usecase it breaks (why is the catch needed)

Reply:
This pull request is related to https://github.com/pointfreeco/swift-tagged/issues/23

I don't remember all the details right now, but I think the problem is that with the container there was an issue with decoding optionals, and using the init directly skipped custom date decoding strategies, so this was the only solution that fixed both issues. You can check this by downloading the repository, commenting each strategy and trying to pass the tests. You will see that in each case one test fails.